homeassistant/components/homeworks.py

```diff
@@ -0,0 +1,151 @@
+"""Component for interfacing to Lutron Homeworks Series 4 and 8 systems.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/homeworks/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.const import (
+    CONF_HOST, CONF_ID, CONF_NAME, CONF_PORT, EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.dispatcher import (dispatcher_send,
+    async_dispatcher_connect)
+from homeassistant.util import slugify
+
+REQUIREMENTS = ['pyhomeworks==0.0.6']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'homeworks'
+
+HOMEWORKS_CONTROLLER = 'homeworks'
+ENTITY_SIGNAL = 'homeworks_entity_{}'
+EVENT_BUTTON_PRESS = 'homeworks_button_press'
+EVENT_BUTTON_RELEASE = 'homeworks_button_release'
+
+CONF_DIMMERS = 'dimmers'
+CONF_KEYPADS = 'keypads'
+CONF_ADDR = 'addr'
+CONF_RATE = 'rate'
+
+FADE_RATE = 1.
+
+CV_FADE_RATE = vol.All(vol.Coerce(float), vol.Range(min=0, max=20))
+
+DIMMER_SCHEMA = vol.Schema({
+    vol.Required(CONF_ADDR): cv.string,
+    vol.Required(CONF_NAME): cv.string,
+    vol.Optional(CONF_RATE, default=FADE_RATE): CV_FADE_RATE
+})
+
+KEYPAD_SCHEMA = vol.Schema({
+    vol.Required(CONF_ADDR): cv.string,
+    vol.Required(CONF_NAME): cv.string,
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT): cv.port,
+        vol.Optional(CONF_DIMMERS): vol.All(cv.ensure_list, [DIMMER_SCHEMA]),
+        vol.Optional(CONF_KEYPADS): vol.All(cv.ensure_list, [KEYPAD_SCHEMA])
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, base_config):
+    """Start Homeworks controller."""
+    from pyhomeworks.pyhomeworks import Homeworks
+
+    class HomeworksController(Homeworks):
+        """Interface between HASS and Homeworks controller."""
+
+        def __init__(self, host, port):
+            """Host and port of Lutron Homeworks controller."""
+            super().__init__(host, port, self.callback)
+
+        def callback(self, msg_type, values):
+            """Dispatch state changes."""
+            _LOGGER.debug('callback: %s, %s', msg_type, values)
+            addr = values[0]
+            signal = ENTITY_SIGNAL.format(addr)
+            dispatcher_send(hass, signal, msg_type, values)
+            _LOGGER.debug('callback returned')
```

Comment:
The callback will return before the signal target is executed. So this debug message doesn't add very much.